solver/src/driver.rs

```diff
@@ -125,38 +126,89 @@ impl Driver {
         &self,
         settlements: Vec<RatedSettlement>,
-    ) -> Result<Vec<RatedSettlement>> {
+    ) -> Result<(Vec<RatedSettlement>, Vec<(RatedSettlement, Error)>)> {
         let simulations = settlement_simulation::simulate_settlements(
             settlements
                 .iter()
                 .map(|settlement| settlement.settlement.clone().into()),
             &self.settlement_contract,
             &self.web3,
             &self.network_id,
+            settlement_simulation::Block::LatestWithoutTenderly,
         )
         .await
         .context("failed to simulate settlements")?;
+
         Ok(settlements
             .into_iter()
             .zip(simulations)
-            .filter_map(|(settlement, simulation)| match simulation {
-                Ok(()) => Some(settlement),
-                Err(err) => {
-                    tracing::error!("settlement simulation failed\n error: {:?}", err);
-                    tracing::debug!("settlement failure for: \n{:#?}", settlement.settlement,);
-                    None
-                }
-            })
-            .collect())
+            .partition_map(|(settlement, result)| match result {
+                Ok(()) => Either::Left(settlement),
+                Err(err) => Either::Right((settlement, err)),
+            }))
+    }
+
+    // Log simulation errors only if the simulation also fails in the block at which on chain
+    // liquidity was queried. If the simulation succeeds at the previous block then the solver
+    // worked correctly and the error doesn't have to be reported.
+    // Note that we could still report a false positive because the earlier block might be off by if
+    // the block has changed just as were were querying the node.
+    async fn report_simulation_errors(
+        &self,
+        errors: Vec<(RatedSettlement, Error)>,
+        current_block_during_liquidity_fetch: u64,
+    ) {
+        let simulations = match settlement_simulation::simulate_settlements(
+            errors
+                .iter()
+                .map(|(settlement, _)| settlement.settlement.clone().into()),
+            &self.settlement_contract,
+            &self.web3,
+            &self.network_id,
+            settlement_simulation::Block::FixedWithTenderly(current_block_during_liquidity_fetch),
+        )
+        .await
+        {
+            Ok(simulations) => simulations,
+            Err(err) => {
+                tracing::error!(
+                    "failed to simulate failing settlements at earlier block {}: {:?}",
+                    current_block_during_liquidity_fetch,
+                    err
+                );
+                return;
+            }
+        };
+
+        for ((settlement, _previous_error), result) in errors.into_iter().zip(simulations) {
+            let error_at_earlier_block = match result {
+                Ok(()) => continue,
+                Err(err) => err,
+            };
+            tracing::error!(
```

Comment:
Maybe we can even "store" the settlement in some form and avoid logging an error multiple times for the same settlement/order?